app/javascript/packs/rooms.js

```diff
@@ -29,6 +29,18 @@ $(document).on('turbolinks:load', function () {
         });
     })
 
+    $('#revoke-code-btn').on('click', function () {
+        var revoke_code_url = $(this).data('url');
+        $.ajax({
+            url: revoke_code_url,
+            type: "POST",
+            beforeSend: function (xhr) { xhr.setRequestHeader('X-CSRF-Token', $('meta[name="csrf-token"]').attr('content')) },
+            data: "",
+        });
+    })
```

Comment:
One final question - why not use `button_to` in this situation? Rather than having to do all this